wandb/sdk/launch/builder/docker.py

```diff
@@ -36,15 +39,33 @@ def build_image(
         repository: Optional[str],
         entrypoint: EntryPoint,
     ) -> str:
+        dockerfile_str = generate_dockerfile(
+            launch_project, entrypoint, launch_project.resource, self.type
+        )
+
+        image_tag = image_tag_from_dockerfile_and_source(launch_project, dockerfile_str)
 
         if repository:
-            image_uri = f"{repository}:{launch_project.image_tag}"
+            image_uri = f"{repository}:{image_tag}"
         else:
-            image_uri = launch_project.image_uri
+            image_uri = f"{launch_project.image_name}:{image_tag}"
```

Comment:
this logic is crazy, can we add a comment?

Reply:
I'll modify this logic when that goes in